geode-core/src/main/java/org/apache/geode/internal/metrics/GeodeCompositeMeterRegistry.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.metrics;
+
+import java.util.Set;
+
+import io.micrometer.core.instrument.binder.MeterBinder;
+import io.micrometer.core.instrument.composite.CompositeMeterRegistry;
+
+public class GeodeCompositeMeterRegistry extends CompositeMeterRegistry {
```

Comment:
Should this class extend `AutoCloseable` since we're making it contain resources that need to be cleaned up?

Reply:
```AutoCloseable``` only closes in a try-with-resources block which this is not. At least, that is my understanding from the javadoc.

Reply:
In my mind, the primary advantage of extending the `AutoCloseable` interface in this case is that it tells the user of this class that the class encapsulates resources that need to be closed. A secondary benefit is that it provides a convenient way to allow this class to be used within a try-with-resources block (although it is not currently used like that, as you pointed out).